### ISSUE TITLE
Subliminal Distillation Audit SchemaPack: provenance for hidden trait transfer and update admission

### ISSUE THREAD

**Issue by crates** [entif-ai/rosetta#1584]
## Summary

Create an experimental **Subliminal Distillation Audit SchemaPack** that can ingest the public run artifacts from van der Weijden, Brack, and Báez Santamaría's reproduction study and emit content-addressed, non-collapsing records for teacher conditioning, carrier generation and filtering, student updates, behavioral evaluations, and audit caveats.

This is a proposed Rosetta integration. The paper authors have not endorsed it.

The first profile should be `subliminal-distillation-audit-v1`. It must preserve the difference between:

- a teacher being conditioned on a trait;
- a carrier output passing a surface-content filter;
- a student being updated from that carrier;
- a behavioral preference shift being measured;
- refusal or identity-leak behavior changing; and
- an operator authorizing downstream use of the updated model.

## Research basis

Paper:

- Daan van der Weijden, Nathan Brack, Selene Báez Santamaría, "Reproducing and Evaluating the Generalizability of Subliminal Learning in Open-Weight Models," arXiv:2609.12586: https://arxiv.org/abs/2609.12586
- Full paper HTML: https://arxiv.org/html/2609.12586

Public proof-of-concept code and artifacts:

- https://github.com/daanvdweijden/subliminal-learning-blackboxnlp2026
- `README.md` documents a three-stage dataset -> LoRA fine-tune -> evaluation pipeline and a stable artifact layout.
- `cfgs/run_cfg.py` declares model, carrier task, trait, answer-space, sampling, filter, fine-tuning, and evaluation configuration.
- `scripts/run_experiment.sh` writes resumable per-cell manifests with model/task/trait/category/seed, source commit, dirty-worktree state, stage status, paths, and durations.
- Public `data/runs/**/manifest.json` records make the adapter testable against real examples.
- The repository states an MIT license and attribution to the upstream implementation.

Relevant Entif and Rosetta sources:

- We Had the Seeds of a Babel Fish: https://entif.ai/tags/research/2026/09/10/babel-fish-alien-mind/
- Rosetta Pasigraphy Protocol: https://entif.ai/tags/research/2026/09/09/rosetta-pasigraphy-protocol/
- Rosetta v3.0.0 Core Spine: `docs/RFCs/Rosetta v3.0.0 Core Spine Specification.md`
- Pack filesystem contract: `docs/RFCs/ROCK-3111-C-RRP-Pack-Filesystem-Contract-v0.2.0.md`
- Pack manifest schema: `packs/_schemas/pack-manifest.schema.json`

Paper-grounded findings that the profile must not flatten:

- trait transmission reproduces in open-weight models but varies substantially by trait, carrier task, and model family;
- Qwen exhibits a stronger effect than Gemma, while Ministral is nearly null in the tested grid;
- shrinking the number answer space from three digits to one strengthens transmission in the reported Qwen ablation;
- the authors propose token-density/concentration as a plausible explanation, not a demonstrated mechanism;
- a logistic probe on one carrier analysis reports AUC near chance, which does not establish the absence of all hidden signal;
- completion health changes are behaviorally material: Qwen shows substantial identity-leak inflation in one animal condition and reduced refusal in politician conditions;
- chess outputs are checked for syntax but not board-state legality;
- teacher-generated data are fixed per trait and shared across fine-tuning seeds;
- confidence intervals are over fine-tuning seeds, and entity-level multiple comparisons are not corrected.

## Why this belongs in Rosetta

The paper exposes a concrete authority gap: surface-valid, semantically unrelated training examples can still carry update-relevant behavioral influence. A dataset's visible denotation, its passed filter result, its learned effect, and its permission for durable model update are different claims.

Rosetta's Core already provides content-addressed Tiles, `Run -> Action -> Observation -> Evaluation -> Receipt`, explicit provenance, and append-only lineage. The pack contract supplies a repository-native way to add domain schemas, profiles, translators, fixtures, and negative tests without changing Core.

This proposal therefore belongs as an experimental SchemaPack and adapter, not as a new theory of subliminal learning and not as a Core semantic mutation.

## Concept and primitive crosswalk

| Research/code construct | Proposed Rosetta representation | Mapping strength / gap |
| --- | --- | --- |
| One model/task/trait/seed cell | `rosetta.run` plus `sda.experiment_run` profile payload | Direct for run identity; add source-commit and environment fields |
| Trait system prompt or control arm | Immutable Observation referenced by `sda.teacher_conditioning` | Direct artifact mapping; does not reveal hidden teacher state |
| Raw 30,000 teacher completions | Dataset-manifest Observation with content/file digests | Direct provenance mapping; large payloads may remain external |
| Filter function and accepted 10,000 examples | `sda.carrier_filter` Action plus input/output dataset refs | Direct; requires explicit filter implementation/version and rejection summary |
| Base, control-FT, and trait-FT arms | Typed `sda.arm` vocabulary | Direct; arms must never be inferred from path text alone |
| LoRA student update | `sda.model_update` Action referencing base checkpoint, dataset, config, seed, and adapter | Direct at declared configuration level; not a faithful neural-mechanism trace |
| Preference probes and completion samples | Evaluation inputs/Observations | Direct |
| Target rate, log-odds ratio, confidence interval | `sda.behavioral_shift_evaluation` | Direct if absolute denominators and comparison arm are retained |
| did_task / refused / other sidecar | `sda.completion_health_evaluation` | Direct; judge identity and prompt/config remain required |
| Carrier logistic-probe AUC | `sda.carrier_probe_evaluation` | Direct metric record; cannot be promoted to "no hidden channel" |
| "Token concentration may explain stronger transfer" | Conjecture linked to the digit-length Evaluation | Partial: proposed explanation, not causal finding |
| Permission to use a student downstream | Separate policy Evaluation / Receipt | Rosetta-side addition; not supplied by the paper |

## Recommended collaboration artifact

Use a **SchemaPack** rooted at:

`packs/subliminal-distillation-audit/`

Rationale: the research implementation already exports structured manifests and predictable artifact paths. The missing integration object is a versioned schema and source-preserving translation profile, not a new runtime, model trainer, or general-purpose safety policy.

A later RecipePack may orchestrate a pre-update audit, but that should depend on evidence from the SchemaPack rather than being bundled into v1.

## Proposed architecture

1. **Source adapter** reads the public experiment manifest and resolved artifact paths without altering source files.
2. **Artifact hasher** creates exact content references for configuration, raw/filtered dataset manifests, model metadata, evaluation outputs, and completion-health sidecars.
3. **Normalizer** emits profile objects for experiment identity, conditioning, carrier generation, filtering, update, and evaluation.
4. **Validator** applies JSON Schema and profile-specific cross-artifact invariants.
5. **Receipt builder** emits an audit receipt describing what was observed and validated, plus missing or unverifiable evidence.
6. **Policy boundary** leaves downstream model admission to a separate consumer. A conformant audit receipt is not an authorization to train, deploy, or rely on the student.

Suggested flow:

```mermaid
flowchart LR
  A[Public run manifest] --> B[Source-preserving adapter]
  C[Config and artifact files] --> B
  B --> D[Content-addressed audit objects]
  D --> E[Schema and invariant validation]
  E --> F[Audit receipt]
  F --> G[Separate downstream admission policy]
```

## Translation and interoperability design

Define at least these schemas:

- `schema/experiment-run.schema.json`
- `schema/teacher-conditioning.schema.json`
- `schema/carrier-dataset.schema.json`
- `schema/carrier-filter.schema.json`
- `schema/model-update.schema.json`
- `schema/behavioral-evaluation.schema.json`
- `schema/completion-health.schema.json`
- `schema/audit-receipt.schema.json`

Define controlled vocabularies for:

- arm: `base`, `control_ft`, `trait_ft`;
- carrier: `numbers`, `numbers_1digit`, `numbers_2digit`, `chess`, `other`;
- evaluation role: `teacher_check`, `preference_shift`, `completion_health`, `carrier_probe`;
- evidence status: `observed`, `declared`, `derived`, `missing`, `not_applicable`;
- caveat: `fixed_teacher_dataset`, `uncorrected_multiple_comparisons`, `relative_metric_baseline_sensitive`, `chess_legality_unverified`, `judge_model_dependent`, `dirty_source_tree`.

The BlackboxNLP adapter should map explicit manifest/config fields first and record unmapped source fields. It must not reconstruct a trait, arm, checkpoint, or filter version from directory names when authoritative metadata are absent.

Interoperability exports may include JSON Schema first. SHACL can follow only if graph-level constraints materially improve cross-artifact validation.

## Repository and implementation impact

Proposed files:

```text
packs/subliminal-distillation-audit/
  pack.json
  README.md
  CHANGELOG.md
  acceptance.md
  metadata.json
  schema/
  vocab/
  profiles/subliminal-distillation-audit-v1.json
  translators/blackboxnlp2026/
  examples/
  test-vectors/
  receipts-fixtures/
  tests/
```

Integrate with:

- `packs/_schemas/pack-manifest.schema.json` for manifest conformance;
- existing pack-ID/conformance tooling required by ROCK-3111-C;
- existing CID helpers for exact source and normalized artifact identity;
- existing receipt schemas by reference where compatible, without redefining receipt semantics.

Do not add new Core tile kinds in v1.

## Phased engineering plan

**Phase 0: evidence pinning and fixture policy**

- Pin paper version, source repository commit, and cited source paths.
- Decide which public fixtures may be copied and which must remain hash/reference-only.
- Record upstream and model/dataset license boundaries.

**Phase 1: schemas and vocabulary**

- Implement the eight schemas and controlled vocabularies.
- Define required cross-references and missing-evidence behavior.
- Add minimal positive and negative examples.

**Phase 2: source adapter**

- Implement a provider-neutral adapter interface.
- Add the `blackboxnlp2026` translator for `manifest.json`, `model.json`, evaluation outputs, and completion-health sidecars.
- Preserve original fields and source hashes.

**Phase 3: validation and receipt**

- Add cross-artifact invariants.
- Emit a receipt with conformance results, caveats, omissions, and source refs.
- Prove that the receipt does not imply mechanism identification or deployment authorization.

**Phase 4: comparison fixtures**

- Add base/control/trait-FT comparison fixtures.
- Add digit-length and model-family comparison fixtures.
- Exercise absolute rate, relative log-odds, completion-health, and carrier-probe records separately.

**Phase 5: optional collaboration pilot**

- With researcher input, map one small public experiment cell end to end.
- Compare normalized outputs with the authors' intended semantics.
- Revise the adapter/profile through a new version rather than rewriting prior artifacts.

## Detailed task checklist

- [ ] Register a unique namespace and conformant `pack.json`.
- [ ] Pin arXiv v1 and a source repository commit.
- [ ] Define source-object and normalized-object identity rules.
- [ ] Define all eight JSON Schemas.
- [ ] Define arm, carrier, evaluation-role, evidence-status, and caveat vocabularies.
- [ ] Preserve teacher prompt or explicit control status as a first-class source reference.
- [ ] Preserve raw and filtered dataset identities, counts, filter version, and rejection summary.
- [ ] Preserve base checkpoint, LoRA configuration, student seed, adapter reference, and code revision.
- [ ] Preserve absolute counts/rates alongside log-odds and confidence intervals.
- [ ] Preserve evaluation prompt family, sampling settings, judge identity, and parse failures.
- [ ] Record whether teacher data were resampled across seeds.
- [ ] Record chess syntax validation separately from board-state legality.
- [ ] Implement source adapter and round-trip/source-retention report.
- [ ] Add positive, tamper, omission, and semantic-overclaim fixtures.
- [ ] Add a no-network CI test profile.
- [ ] Document that conformance is not update or deployment authority.
- [ ] Run pack-ID and repository conformance checks.
- [ ] Invite paper authors to review field meanings and the pilot mapping.

## Acceptance criteria and verification

- A valid public run manifest plus referenced metadata produces a deterministic normalized bundle and audit receipt.
- Changing any source artifact changes the relevant CID and invalidates stale derived receipts.
- Raw and filtered dataset references cannot collapse into one object.
- Base, control-FT, and trait-FT arms cannot be conflated.
- A missing teacher-conditioning record, filter version, comparison arm, or model checkpoint yields explicit incomplete evidence rather than inferred data.
- Relative log-odds cannot validate without its baseline and absolute counts/rates.
- Carrier AUC near chance cannot emit a "no hidden signal" verdict.
- Chess syntax validation cannot emit a chess-legality verdict.
- A confidence interval over fine-tuning seeds records that the teacher dataset was fixed when applicable.
- Completion-health changes remain separate from target-preference changes.
- Dirty source-tree status is retained as a caveat rather than silently dropped.
- The pack passes repository pack-manifest, pack-ID, schema, fixture, and no-network checks.
- No emitted receipt authorizes model update or deployment.

## Risks, constraints, and licensing

- The paper demonstrates behavioral correlations, not the causal encoding mechanism.
- The token-concentration explanation is a hypothesis requiring targeted experiments.
- Results cover three open-weight model families, two carrier tasks, limited trait categories, and one principal answer-space ablation.
- Fixed teacher datasets mean reported seed intervals do not capture teacher-generation variance.
- Uncorrected entity-level comparisons must not be overinterpreted.
- Judge-model classifications can be wrong and must retain judge identity and raw outputs.
- Public manifests may mark `git_dirty: true`; source commit alone may not fully identify the executed tree.
- The code is MIT-licensed, but model checkpoints, generated datasets, prompts, and dependencies may carry separate terms. Do not redistribute large artifacts until each license is checked.
- Trait work involving public figures needs careful framing; the pack describes experiments and must not become a reputation-scoring system.

## Open questions for the researchers

- Which manifest/config fields do you consider authoritative when path naming and runtime metadata disagree?
- Can filter rule identity be pinned to a callable hash or version, not only a source commit?
- Would you expose rejection-reason counts per carrier dataset to distinguish surface validity from selection pressure?
- Which absolute-count fields best preserve the denominator behind the reported log-odds ratios?
- How should teacher-generation uncertainty be represented when datasets are fixed across fine-tuning seeds?
- Is a legal-move checker planned for the chess carrier, and should existing runs retain an explicit `legality_unknown` value?
- Which completion-health raw outputs can be safely included in a compact public pilot fixture?
- What follow-up would most directly discriminate token concentration from alternative optimization or template explanations?

## Non-goals

- Claiming that structured provenance prevents subliminal learning.
- Detecting arbitrary hidden channels from syntax alone.
- Exposing or reconstructing private model internals.
- Reproducing all 1,094 reported experiment cells in Rosetta CI.
- Redistributing model weights or large generated datasets.
- Replacing the authors' analysis code or statistical methodology.
- Treating preference shifts as arbitrary instruction transmission.
- Granting training, deployment, safety, or truth authority from one conformant receipt.
- Modifying Rosetta Core semantics.

## Collaboration path

1. Land schemas, vocabularies, and synthetic fixtures under an experimental namespace.
2. Implement the adapter against one public model/task/trait/seed cell.
3. Ask the paper authors to review the semantic mapping, especially arms, filter identity, denominators, and caveats.
4. Add one public digit-length comparison and one model-family contrast.
5. Publish an acceptance report that lists preserved fields, losses, missing evidence, and rejected overclaims.
6. Promote beyond experimental status only after independent replay and author feedback; lack of author participation should not be represented as endorsement.

## Source references

- Paper abstract and front matter: https://arxiv.org/abs/2609.12586
- Full paper: https://arxiv.org/html/2609.12586
- Public implementation: https://github.com/daanvdweijden/subliminal-learning-blackboxnlp2026
- Implementation README: https://github.com/daanvdweijden/subliminal-learning-blackboxnlp2026/blob/main/README.md
- Run configuration: https://github.com/daanvdweijden/subliminal-learning-blackboxnlp2026/blob/main/cfgs/run_cfg.py
- Pipeline and manifest writer: https://github.com/daanvdweijden/subliminal-learning-blackboxnlp2026/blob/main/scripts/run_experiment.sh
- Rosetta repository: https://github.com/entif-ai/rosetta
- Rosetta pack contract: https://github.com/entif-ai/rosetta/blob/main/docs/RFCs/ROCK-3111-C-RRP-Pack-Filesystem-Contract-v0.2.0.md
- Rosetta pack manifest schema: https://github.com/entif-ai/rosetta/blob/main/packs/_schemas/pack-manifest.schema.json
- We Had the Seeds of a Babel Fish: https://entif.ai/tags/research/2026/09/10/babel-fish-alien-mind/
- Rosetta Pasigraphy Protocol: https://entif.ai/tags/research/2026/09/09/rosetta-pasigraphy-protocol/
